app/models/alchemy/picture_version.rb

```diff
@@ -1,32 +1,50 @@
-class Alchemy::PictureVersion < ActiveRecord::Base
-  has_many :alchemy_dragonfly_signatures, class_name: 'Alchemy::DragonflySignature', foreign_key: :signature,
-                                          primary_key: :signature, dependent: :destroy
-  attr_accessor :image
-
-  def self.from_cache(image)
-    record = self.find_by_signature(image.signature)
-    record.nil? ? Alchemy::PictureVersion.new(image: image) : (record.image = image; record)
-  end
+# frozen_string_literal: true
+
+module Alchemy
+  class PictureVersion < ActiveRecord::Base
+    has_one :dragonfly_signature, foreign_key: :signature, primary_key: :signature
+    attr_writer :image
+
+    before_destroy :remove_from_store
+
+    def self.from_cache(image)
+      record = find_by_signature(image.signature)
```

Comment:
This could be rewritten for concision as:
```ruby
(find_by_signature(image.signature) || new).tap do |record|
  record.image = image
end
```

Reply:
Yeah, there's a slight difference in when .image= is called for new records, but I can't see any issues this would cause for how we're using the PictureVersions.